plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java

```diff
@@ -180,6 +196,69 @@ public void onFailure(Exception e) {
         }
     }
 
+    private void deployAndPredictRemoteModel(
+        String modelId,
+        ActionListener<MLTaskResponse> wrappedListener,
+        MLPredictionTaskRequest mlPredictionTaskRequest
+    ) {
+        String[] workerNodes = mlModelManager.getWorkerNodes(modelId, FunctionName.REMOTE, true);
+        if (!enableAutomaticDeployment || (mlModelManager.isModelDeployed(modelId) && workerNodes != null && workerNodes.length != 0)) {
+            // model deployment finished, or automatic deployment disallowed
+            executePredict(mlPredictionTaskRequest, wrappedListener, modelId);
+            return;
+        }
+        mlModelManager.getModel(modelId, ActionListener.wrap(mlModel -> {
+            // Obtain the deployment plan
+            String[] workerNodeIds = mlModel.getPlanningWorkerNodes();
+            MLDeployModelRequest deployModelRequest = MLDeployModelRequest
+                .builder()
+                .modelId(modelId)
+                .async(false)
+                .dispatchTask(false)
+                .modelNodeIds(workerNodeIds)
+                .build();
+            deployAndPredict(modelId, mlModel, deployModelRequest, mlPredictionTaskRequest, wrappedListener);
+        }, exception -> {
+            log.error("Failed to find the remote model in the model index " + modelId, exception);
+            wrappedListener.onFailure(exception);
+        }));
+    }
+
+    private void deployAndPredict(
+        String modelId,
+        MLModel mlModel,
+        MLDeployModelRequest deployModelRequest,
+        MLPredictionTaskRequest mlPredictionTaskRequest,
+        ActionListener<MLTaskResponse> wrappedListener
+    ) {
+        ActionListener<MLDeployModelResponse> deployModelActionListener = ActionListener.wrap(deployModelResponse -> {
+            // Deployment failed, existing
+            if (!deployModelResponse.getStatus().equals(MLTaskState.COMPLETED.name())) {
+                wrappedListener
+                    .onFailure(
+                        new IllegalArgumentException(
+                            "Deployment Failed for " + modelId + ", please check your remote model before predicting."
+                        )
+                    );
+                return;
+            }
+            String[] workerNodes = mlModelManager.getWorkerNodes(modelId, FunctionName.REMOTE, true);
+            if (workerNodes == null || workerNodes.length == 0) {
+                // The deployment is kicked off, but not finished yet in this node
+                mlModelManager.deployRemoteModelToLocal(modelId, mlModel, ActionListener.wrap(r -> {
```

Comment:
Check if local node is eligible for deploy model first. For example we should not deploy model on master node

Reply:
Good point, we can reuse the current eligible nodes from the ModelManager to check. Will add this in the next refresh.